cpp/src/partitioning/partitioning.cu

```diff
@@ -693,7 +693,7 @@ struct dispatch_map_type {
 
     // Unfortunately need to materialize the scatter map because
     // `detail::scatter` requires multiple passes through the iterator
-    rmm::device_uvector<MapType> scatter_map(partition_map.size(), stream);
+    rmm::device_uvector<size_type> scatter_map(partition_map.size(), stream);
```

Comment:
If we can use `size_type`, we can construct a `column_view` from it and use the `scatter` API taking that `column_view` map instead (line 710 below).

Reply:
Yeah, after changing the header like that then just this:
```
    auto scattered =
-      cudf::detail::scatter(t, scatter_map.begin(), scatter_map.end(), t, stream, mr);
+     cudf::detail::scatter(t, scatter_map, t, stream, mr);
```